xapian-letor/configure.ac

```diff
@@ -569,3 +569,12 @@ AC_CONFIG_FILES([
 AC_CONFIG_FILES([makemanpage], [chmod +x makemanpage])
 AC_CONFIG_FILES([tests/runtest], [chmod +x tests/runtest])
 AC_OUTPUT
+
+dnl There are no files generated by AC_OUTPUT in the following directories
+dnl and we need to ensure they exist so that the rest of configure or make
+dnl won't fail because they don't exist when srcdir != builddir.
+if test yes = "$vpath_build" ; then
+  for dir in include/xapian-letor ; do
+    AS_MKDIR_P(["$dir"])
+  done
```

Comment:
The loop seems like a bit of premature generality - I wouldn't expect us to grow more such directories.

But wouldn't it make more sense to handle this in `generate_exceptions`, since that's what requires this to exist?  That encapsulates the requirement along with where it's required, and is more robust (the user can even `rm -rf include/xapian-letor` after running `configure` and it should still get recreated), and slightly lazier (since we don't run any extra code in a VPATH build from a tarball, where the generated files are already present at configure time).

Reply:
Yes, that's a good idea — I just copied stuff from -core, which has more directories (hence the loop) but also should move that into `generate_exceptions`.

Reply:
Oh, except in -core it can't because immediately after the loop, it builds `include/xapian/version.h`. I think ensuring that the output directory exists in `generate_exceptions` is correct even there, though. I'll add another commit…

Reply:
That's not going to work, because I didn't test it at my end. Sigh.

Reply:
Ah right - yes, I'd agree ensuring the output directory exists in the script that uses it is the better approach generally.